### PR TITLE
New/updated python packages related to the TREZOR hardware wallet

### PR DESCRIPTION
**Pull request by np**

### CODE REVIEW

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -24271,4 +24271,167 @@ in modules // {
...
+      test -d $libusb || { echo "ERROR: $libusb doesn't exist, please update/fix this build expression."; exit 1; }
+      sed -i -e "s|/usr/include/libusb-1.0|$libusb|" setup.py
+    '';
+
+    meta = {
+      description = "A Cython interface to the hidapi from https://github.com/signal11/hidapi";
+      homepage = https://github.com/trezor/cython-hidapi;
+      # license can actually be either bsd3 or gpl3
+      # see https://github.com/trezor/cython-hidapi/blob/master/LICENSE-orig.txt
+      license = licenses.bsd3;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: you can give a list of licenses

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -24271,4 +24271,167 @@ in modules // {
...
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/t/trezor/${name}.tar.gz";
+      sha256 = "0nqbjj0mvkp314hpq36px12hxbyidmhsdflq3121l4g9y3scfbnx";
+    };
+
+    propagatedBuildInputs = with self; [ protobuf2_6 hidapi ];
+
+    buildInputs = with self; [ ecdsa mnemonic ];
+
+    doCheck = false;
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you explain why you disabled tests?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -24271,4 +24271,167 @@ in modules // {
...
+      url = "https://pypi.python.org/packages/source/k/keepkey/${name}.tar.gz";
+      sha256 = "1ikyp4jpydskznsrlwmxh9sn7b64aldwj2lf0phmb19r5kk06qmp";
+    };
+
+    propagatedBuildInputs = with self; [ protobuf2_6 hidapi ];
+
+    buildInputs = with self; [ ecdsa mnemonic ];
+
+    doCheck = false;
+
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: could you explain why you disabled tests?

**Comment by joachifm** [comment]
See also https://github.com/NixOS/nixpkgs/pull/7110

**Comment by FRidh** [comment]
Looks good! Did you test whether all Python packages work with all interpreters? (`nox-review wip` or `nox-review pr 12530`)

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (np) [inline pkgs/top-level/python-packages.nix]: There are no actual tests: "ImportError: No module named tests"

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (np) [inline pkgs/top-level/python-packages.nix]: Same here, keepkey is actually a fork of trezor.

File: pkgs/top-level/python-packages.nix (hunk 2 above)
Comment (FRidh) [inline pkgs/top-level/python-packages.nix]: Please add it as a comment to the code.

**Comment by np** [comment]
I have trouble running nox-review, I'm getting issues such as:

```
error: Your local changes to the following files would be overwritten by merge:
	pkgs/top-level/all-packages.nix
	pkgs/top-level/python-packages.nix
```

**Comment by FRidh** [comment]
Which nixpkgs repository do you edit? If nox gives this error I would think you work on ~/.nox/nixpkgs, which you should not.

**Comment by np** [comment]
I was not working there, however the working copy was dirty, I guess a bug in `nox` or something stopped it half-way. Is the working copy in `~/.nox` really necessary? If so, what about some `git reset --hard XXX` at the beginning.

With the latest commit I could run `nox-review pr 12530` with success. I guess all is good now.

I think the potential overlap with #7110 should be dealt with afterwards.

**Comment by FRidh** [comment]
> I was not working there, however the working copy was dirty, I guess a bug in nox or something stopped it half-way. Is the working copy in ~/.nox really necessary? If so, what about some git reset --hard XXX at the beginning.

Nox uses the copy in `~/.nox`. If it somehow gets broken I always just `rm -rf .nox`.

> With the latest commit I could run nox-review pr 12530 with success. I guess all is good now.

That's good. Let's check again with `nox-review` after https://github.com/NixOS/nixpkgs/pull/12552 is in.

> I think the potential overlap with #7110 should be dealt with afterwards.

I don't know enough about the application you want to get in to judge that; I'm just looking at the Python packages ;-)

**Pull request merged**